Sources/StreamChat/Repositories/PollsRepository.swift

```diff
@@ -201,6 +201,20 @@ class PollsRepository {
         }
     }
     
+    func deletePoll(
+        pollId: String,
+        completion: ((Error?) -> Void)? = nil
+    ) {
+        database.write { session in
+            _ = try? session.deletePoll(pollId: pollId)
+        } completion: { [weak self] _ in
+            // No need to check for error, since the poll can also be deleted independently.
+            self?.apiClient.request(endpoint: .deletePoll(pollId: pollId)) {
+                completion?($0.error)
+            }
+        }
```

Comment:
If we are not doing a rollback here, better to delete the poll only after a successful call, otherwise we might have invalid state. If the deletion call of the poll fails, the user will get the feedback the initially was deleted, but then when querying the channel or polls, it will appear again 🤔

Reply:
makes sense, will update it